agent/exec/container/container.go

```diff
@@ -83,9 +83,11 @@ func (c *containerConfig) image() string {
 }
 
 func (c *containerConfig) config() *enginecontainer.Config {
+	var hostname = strings.Replace(c.spec().Hostname, "*", fmt.Sprint(c.task.Slot), 1)
```

Comment:
> Templated hostname will be useful for me if 'source' parameter of 'mount' option of docker service create is also templated.

This is why templating needs more thought. I'd like this to work consistently and safely across the board.

Yes, this PR will be acceptable without the templating.